homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,639 @@
+
+
+def profile_entities_list(
+    fritzbox_tools: FritzBoxTools, device_friendly_name: str
+) -> list[FritzBoxProfileSwitch]:
+    """Get list of profile entities."""
+    _LOGGER.debug("Setting up %s switches", SWITCH_TYPE_DEVICEPROFILE)
+    if len(fritzbox_tools.fritz_profiles) <= 0:
+        _LOGGER.debug("The FRITZ!Box has no %s options", SWITCH_TYPE_DEVICEPROFILE)
+        return []
+
+    return [
+        FritzBoxProfileSwitch(fritzbox_tools, device_friendly_name, profile)
+        for profile in fritzbox_tools.fritz_profiles.keys()
+    ]
+
+
+def wifi_entities_list(
+    fritzbox_tools: FritzBoxTools, device_friendly_name: str
+) -> list[FritzBoxWifiSwitch]:
+    """Get list of wifi entities."""
+    _LOGGER.debug("Setting up %s switches", SWITCH_TYPE_WIFINETWORK)
+    std_table = {"ac": "5Ghz", "n": "2.4Ghz"}
+    networks: dict = {}
+    for i in range(4):
+        if ("WLANConfiguration" + str(i)) in fritzbox_tools.connection.services:
+            network_info = service_call_action(
+                fritzbox_tools, "WLANConfiguration", str(i), "GetInfo"
+            )
+            if network_info:
+                ssid = network_info["NewSSID"]
+                if ssid in networks.values():
+                    networks[i] = f'{ssid} {std_table[network_info["NewStandard"]]}'
+                else:
+                    networks[i] = ssid
+
+    return [
+        FritzBoxWifiSwitch(fritzbox_tools, device_friendly_name, net, networks[net])
+        for net in networks
+    ]
+
+
+def all_entities_list(
+    fritzbox_tools: FritzBoxTools, device_friendly_name: str
+) -> list[Entity]:
+    """Get a list of all entities."""
+    return [
+        *deflection_entities_list(fritzbox_tools, device_friendly_name),
+        *port_entities_list(fritzbox_tools, device_friendly_name),
+        *profile_entities_list(fritzbox_tools, device_friendly_name),
+        *wifi_entities_list(fritzbox_tools, device_friendly_name),
+    ]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up entry."""
+    _LOGGER.debug("Setting up switches")
+    fritzbox_tools: FritzBoxTools = hass.data[DOMAIN][entry.entry_id]
+
+    _LOGGER.debug("Fritzbox services: %s", fritzbox_tools.connection.services)
+
+    entities_list = await hass.async_add_executor_job(
+        all_entities_list, fritzbox_tools, entry.title
+    )
+    async_add_entities(entities_list)
+
+
+class FritzBoxBaseSwitch(FritzBoxBaseEntity):
+    """Fritz switch base class."""
+
+    def __init__(
+        self,
+        fritzbox_tools: FritzBoxTools,
+        device_friendly_name: str,
+        switch_info: SwitchInfo,
+    ) -> None:
+        """Init Fritzbox port switch."""
+        super().__init__(fritzbox_tools, device_friendly_name)
+
+        self._description = switch_info["description"]
+        self._friendly_name = switch_info["friendly_name"]
+        self._icon = switch_info["icon"]
```

Comment:
Pretty sure _attr_icon will work directly

Reply:
as per discord chat, we skip this for now